### ISSUE TITLE
Issue with resource Google BigQuery, Tables are getting re-created

### ISSUE THREAD

**Issue by Dhanabalan-pop** [hashicorp/terraform-provider-google#8338]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

We are using docker  image: docker:18.09
Provider versions:
provider.google-beta: version = "~> 3.54"
So it should be latest terraform

### Affected Resource(s)

*google_bigquery_table

### Terraform Configuration Files

```tf
resource "google_bigquery_table" "TEST" {
  project = var.eim_project
  dataset_id = var.core_dataset_name
  table_id   = "TEST"
  schema = file("modules/resources/schemas/test.schema.json")
}

```

### Debug Output

 # module.resources.google_bigquery_table.TEST must be replaced

~ {
                    name = "tests"
                  ~ type = "STRING" -> "String"
                } # forces replacement,
              ~ {
                    name = "test"
                  ~ type = "STRING" -> "String"
                } # forces replacement,

### Panic Output

Deleted table and re-created

### Expected Behavior

It shouldn't delete

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by venkykuberan** [comment]
@Dhanabalan-pop can you attach your test schema file to repro the issue.

**Comment by rileykarson** [comment]
See https://github.com/hashicorp/terraform-provider-google/issues/8337 for context, we're aware and details are outlined. There should be a fix out in `3.55.0` on Monday.

**Comment by Dhanabalan-pop** [comment]
terraform plan output:
~ {
                  + mode = "NULLABLE"
                    name = "Test3"
                    type = "INTEGER"
                } # forces replacement,
              ~ {
                  + mode = "NULLABLE"
                    name = "Test2"
                    type = "STRING"
                } # forces replacement,
              ~ {
                  + mode = "NULLABLE"
                  ~ name = "Test" -> "Test1"
                  ~ type = "DATE" -> "INTEGER"
                } # forces replacement,
              ~ {
                  + mode = "NULLABLE"
                  ~ name = "Test1" -> "Test"
                  ~ type = "INTEGER" -> "DATE"
                } # forces replacement,
                Schema:
                    "mode": "NULLABLE", 
    "name": "Test3", 
    "type": "INTEGER"
  }, 
  {
    "mode": "NULLABLE", 
    "name": "Test2", 
    "type": "STRING"
  }, 
  {
    "mode": "NULLABLE", 
    "name": "Test", 
    "type": "INTEGER"
  }, 
  {
    "mode": "NULLABLE", 
    "name": "Test1", 
    "type": "DATE"
  },
